cmake/inputs/patches/ep_gcssdk/build.patch

```diff
@@ -0,0 +1,377 @@
+From a06cbecee7a62b3aec451e61ae1d8c3c8b73805e Mon Sep 17 00:00:00 2001
+From: Joe maley <joe@tiledb.io>
+Date: Wed, 11 Mar 2020 11:26:06 -0400
+Subject: GCS Build Patch for TileDB
+
+---
+ cmake/FindCurlWithTargets.cmake              | 25 +++++++-
+ super/CMakeLists.txt                         | 10 ++-
+ super/external/c-ares.cmake                  |  2 +
+ super/external/crc32c.cmake                  |  2 +
+ super/external/curl.cmake                    | 64 --------------------
+ super/external/google-cloud-cpp-common.cmake |  2 +
+ super/external/googleapis.cmake              |  2 +
+ super/external/googletest.cmake              |  2 +
+ super/external/grpc.cmake                    |  5 +-
+ super/external/protobuf.cmake                |  4 +-
+ super/external/ssl.cmake                     | 26 --------
+ super/external/zlib.cmake                    | 47 --------------
+ 12 files changed, 46 insertions(+), 145 deletions(-)
+ delete mode 100644 super/external/curl.cmake
+ delete mode 100644 super/external/ssl.cmake
+ delete mode 100644 super/external/zlib.cmake
+
+diff --git a/cmake/FindCurlWithTargets.cmake b/cmake/FindCurlWithTargets.cmake
+index 5fe8cbfba..760396983 100644
+--- a/cmake/FindCurlWithTargets.cmake
++++ b/cmake/FindCurlWithTargets.cmake
+@@ -30,7 +30,30 @@ else ()
+     # As searching for libcurl using CONFIG mode failed, try again using the
+     # CMake config module. We will need to fix up a few things if the module is
+     # found this way.
+-    find_package(CURL REQUIRED)
```

Comment:
@ihnorton This is new -- I had to tell the GCS build where to find curl in our install directory.

Reply:
setting `CURL_ROOT_DIR` should help

Reply:
I think the issue here is that if CURL has to be built, we don't know its location at configure-time to pass it to the GCS superbuild

Reply:
We do know because if we are building it from source it'll be in the `TILEDB_EP_INSTALL_PREFIX`, so the `-DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}` will allow it to find curl/openssl/anything built from TileDB super build. GCS externalproject isn't called until the build phase, so by that time (because of the `       list(APPEND DEPENDS ep_openssl)`), we'll have build openssl/curl and it'll be discovered through cmake `find_package`.

Reply:
yep. scratch the `CURL_ROOT_DIR` though, I forgot that the builtin `FindCurl` doesn't use that, so you need to either set `CMAKE_PREFIX_PATH` or (probably) the pair of `-DCURL_INCLUDE_DIR` and `-DCURL_LIBRARY`.

Reply:
I've passed the `TILEDB_EP_INSTALL_PREFIX` cmake var to the GCS external project. The GCS external project picks it up and appends it to the `CMAKE_PREFIX_PATH` and passes it through its nested external projects. I've removed the code at this comment, so we're back to the unpatched code to find curl simply through `find_package`.